### PR TITLE
Fixes dropping item in container 

### PR DESCRIPTION
**Pull request by Plykiya**
## About the PR

Dropping an item while inside a container will now place the item inside the container, rather than outside of it

## Why / Balance

Fixes https://github.com/space-wizards/space-station-14/issues/29624
Inspired by previous fix attempt https://github.com/space-wizards/space-station-14/pull/28167

## Technical details

The previous implementation would attempt to drop the item twice, the second instance placing the item outside the container. I changed the order of interactions so that it only drops the item once.

- If the player is inside a container, it will use the transformSystem's DropNextTo() method which successfully accounts for dropping items into containers.
- If the player is outside a container, the drop location is calculated based on the direction they click and their interaction range, and the item is placed at the calculated location.

As requested by the other PR I added an integration test that apparently works when ran, so uh, I guess that's good. It's my first time implementing them for Space Station 14 so I just barely understand how it works. I won't lie, I committed some heinous copy pasta sins 

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/space-wizards/space-station-14/assets/58439124/cd888ff4-4199-4bc7-a328-13d22e3244b3

image

**Changelog**

:cl:
- fix: Dropping an item while in a container now places the item in the container.

### CODE REVIEW

**Comment by Plykiya** [comment]
yeah that test fail ain't mine

**Comment by lzk228** [comment]
maybe you also can look #26061 ? i suppose its quite similar

**Comment by Plykiya** [comment]
> maybe you also can look #26061 ? i suppose its quite similar

yeah I'm going to do stripping issues next

File: Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs [hunk 1]
```diff
@@ -122,18 +125,23 @@ public bool TryDrop(EntityUid uid, Hand hand, EntityCoordinates? targetDropLocat
...
-        var origin = new MapCoordinates(itemPos, itemXform.MapID);
-        var target = targetDropLocation.Value.ToMap(EntityManager, TransformSystem);
-        TransformSystem.SetWorldPositionRotation(entity, GetFinalDropCoordinates(uid, origin, target), itemRot);
-        return true;
+            var (itemPos, itemRot) = TransformSystem.GetWorldPositionRotation(entity);
+            var origin = new MapCoordinates(itemPos, itemXform.MapID);
+            var target = TransformSystem.ToMapCoordinates(targetDropLocation.Value);
+            TransformSystem.SetWorldPositionRotation(entity, GetFinalDropCoordinates(uid, origin, target), itemRot);
+            return true;
+        }
```
Comment (Brandon-Huu) [inline Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs:144]: Shouldn't you remove the else block (keeping the code).
or move the return statements after the if-else blocks?

Comment (Plykiya) [reply Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs:144]: it's functionally no different, it always go to either the if or else, but I can move it outside to reduce the line difference

**Pull request merged**